aws-kendra-datasource/aws-kendra-datasource.json

```diff
@@ -902,9 +903,7 @@
   "required": [
     "Name",
     "IndexId",
-    "Type",
-    "DataSourceConfiguration",
-    "RoleArn"
```

Comment:
For CUSTOM Datasource type DataSourceConfiguration and RoleArn is not required in the latest kendra SDK 
https://docs.aws.amazon.com/kendra/latest/dg/API_CreateDataSource.html

Reply:
How do we make sure they are provided for other configurations? Please add code to make sure RoleArn & DataSourceConfiguration are present if type != Custom.

Reply:
Hi Eric, so this section of the document would be updated once i work with the cloudformation team & chuck on updating the cloudformation documentation for datasource. 
We'll explicitly mention user is not suppose to provide RoleArn, DataSourceConfiguration if creating "CUSTOM" datasource, just like how we've mentioned here in the SDK release for"CUSTOM"datasource
https://docs.aws.amazon.com/kendra/latest/dg/API_CreateDataSource.html
Please look at the section "CONFIGURATION" and "ROLEARN"

Reply:
I mean this section of the document would be updated : 
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-kendra-datasource.html

Reply:
Discussed offline, and the service layer response will be sent to customer.  Testing of service layer is outside scope of this CR.